_targets.R

```diff
@@ -6,11 +6,44 @@ source("3_visualize/src/plot_timeseries.R")
 options(tidyverse.quiet = TRUE)
 tar_option_set(packages = c("tidyverse", "dataRetrieval")) # Loading tidyverse because we need dplyr, ggplot2, readr, stringr, and purrr
 
+# define NWIS stations of interest and file out paths
+nwis_stn <- c("01427207", "01432160", "01436690", "01466500") # removing "01435000" because it returns an unexpected column name
+fileout_stn <- paste0("1_fetch/out/nwis_", nwis_stn, ".csv")
+nwis_target <- paste0("nwis_", nwis_stn)
+
 p1_targets_list <- list(
+  # download data
+  tar_target(
+    nwis_target[1],
+    download_nwis_data(site_no = nwis_stn[1], pathout = "1_fetch/out/"),
+    format = "file"
+  ),
+  
+  tar_target(
+    nwis_target[2],
+    download_nwis_data(site_no = nwis_stn[2], pathout = "1_fetch/out/"),
+    format = "file"
+  ),
+  
+  tar_target(
+    nwis_target[3],
+    download_nwis_data(site_no = nwis_stn[3], pathout = "1_fetch/out/"),
+    format = "file"
+  ),
+  
+  tar_target(
+    nwis_target[4],
+    download_nwis_data(site_no = nwis_stn[4], pathout = "1_fetch/out/"),
+    format = "file"
+  ),
+
+  # combine into one data set
   tar_target(
     site_data,
-    download_nwis_data(),
+    combine_nwis_data("1_fetch/out")
   ),
```

Comment:
I like this thought process on essentially having a your pipeline check for what is in a folder as it can scale well. Couple notes on this - targets actually can use a directory as a target which works great for a situation where you have a folder and files might get periodically added that you would then want the pipeline to rebuild. In pipelines-3 training you'll also go over dynamic branching which will demonstrate how you can scale to lots of sites easily. But here with a static pipeline the downside to your approach here is that the individual site targets aren't explicitly linked (as inputs) to the `combine_nwis_data` function target. This can cause the pipeline not to rebuild the correct targets sometimes. So here, what I would suggest is to change this to pass the individual site targets into your combine function directly in the targets make file. This approach wouldn't scale well but you should see it makes the dependencies nice and clear.